packages/design-picker/src/components/unified-design-picker.tsx

```diff
@@ -62,6 +64,83 @@ const DesignPreviewImage: React.FC< DesignPreviewImageProps > = ( {
 	);
 };
 
+interface TrackDesignViewProps {
+	category?: string | null;
+	design: Design;
+	isPremiumThemeAvailable?: boolean;
+}
+
+/**
+ * Hook to return a [callback ref](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs)
```

Comment:
The drawback of this way is the component using this hook might not able to get the ref if they need it. For example,

```js
const DesignButtonContainer = () => {
  const trackingDivRef = useTrackDesignView( ... );

  useEffect( () => {
    trackingDivRef.current.focus(); // ❌ because it's not a ref
  }, [ ] );

  return <div ref={ trackingDivRef }>...</div>
}
```

But we could consider this later when we have this requirement.

Reply:
I agree that we can defer this kind of change. I _think_ we're unlikely to need a ref to the `div` for anything, but I could easily be wrong about that. Still, we can tackle that issue when we need to.